File: comparisons/parity/sleepstudy_offset.jl
# LMM with offset parity exporter (MixedModels.jl).
#   julia comparisons/parity/sleepstudy_offset.jl

using CSV
using DataFrames
using JSON
using MixedModels

file_path = joinpath("tests", "data", "sleepstudy.csv")
isfile(file_path) || error("missing dataset: $file_path")

df = CSV.read(file_path, DataFrame)
df.Subject = string.(df.Subject)

form = @formula(Reaction ~ 1 + Days + offset(OffsetDays10) + (Days | Subject))
m = fit(MixedModel, form, df, REML = true)

coef_names = coefnames(m)
coef_vals = coef(m)
coefficients = Dict(zip(coef_names, coef_vals))

payload = Dict(
    "case" => "sleepstudy_offset",
    "implementation" => "julia",
    "coefficients" => coefficients,
)
println(JSON.json(payload))

File: comparisons/parity/sleepstudy_offset.R
source("comparisons/parity/parity_r_setup.R")
suppressPackageStartupMessages({
  library(lme4)
  library(jsonlite)
})
source("comparisons/r_setup.R")

data <- read.csv("tests/data/sleepstudy.csv")
data$Subject <- as.factor(data$Subject)
fit <- lmer(
  Reaction ~ Days + offset(OffsetDays10) + (Days | Subject),
  data = data,
  REML = TRUE
)
coefs <- as.list(as.numeric(fixef(fit)))
names(coefs) <- names(fixef(fit))
cat(toJSON(
  list(case = "sleepstudy_offset", implementation = "r", coefficients = coefs),
  auto_unbox = TRUE,
  digits = 10
))
